mmv1/third_party/terraform/fwresource/with_get_region.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package fwresource
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+type RegionGetter interface {
+	GetRegion(types.String, *transport_tpg.Config, *diag.Diagnostics) types.String
+}
+
+// WithGetRegion can be embedded into structs implementing resource and data source interfaces
+// that are expected to contain a Region field
+type WithGetRegion struct {
+	Config *transport_tpg.Config
+}
```

Comment:
See this comment on another PR: https://github.com/GoogleCloudPlatform/magic-modules/pull/11925#discussion_r1853945991

This cannot be embedded in model structs